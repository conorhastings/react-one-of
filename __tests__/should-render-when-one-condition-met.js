import React from 'react';
import renderer from 'react-test-renderer';
import OneOf from "../src/index.js";

function X() {
	return "x";
}

function Y() {
	return "y";
}

function Z() {
	return "z";
}

test('OneOf component renders correctly when only one condition is met', () => {
  const tree = renderer.create(
    <OneOf
    	components={[
    		{ Component: X, condition: true },
    		{ Component: Y, condition:false },
    		{ Component: Z, condition: false }
    	]}
    	cool="wow"
    	conor="awesome"
    	volkswagen="evil"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});