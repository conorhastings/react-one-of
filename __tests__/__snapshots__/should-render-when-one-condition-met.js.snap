// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneOf component renders correctly when only one condition is met 1`] = `"x"`;
